app/lwip/core/dns.c

```diff
@@ -929,6 +929,9 @@ dns_enqueue(const char *name, dns_found_callback found, void *callback_arg)
   pEntry->found = found;
   pEntry->arg   = callback_arg;
   namelen = LWIP_MIN(os_strlen(name), DNS_MAX_NAME_LENGTH-1);
+  if ((pEntry->name = (char *) mem_realloc(pEntry->name, namelen+1)) == NULL) {
+    return ERR_MEM;
```

Comment:
Perhaps worth cleaning out the rest of `pEntry` on this path (e.g., setting `found` and `callback_arg` to `NULL`)?